fuel-tx/src/tests/valid_cases/transaction.rs

```diff
@@ -834,39 +834,53 @@ fn mint() {
 
     let block_height = 1000.into();
 
-    TransactionBuilder::mint(block_height, rng.gen())
-        .add_output(Output::coin(rng.gen(), rng.next_u64(), rng.gen()))
-        .add_output(Output::coin(rng.gen(), rng.next_u64(), rng.gen()))
-        .finalize()
-        .check(block_height, &test_params())
-        .expect("Failed to validate tx");
-
-    let err = TransactionBuilder::mint(block_height, rng.gen())
-        .add_output(Output::contract(0, rng.gen(), rng.gen()))
-        .finalize()
-        .check(block_height, &test_params())
-        .expect_err("Expected erroneous transaction");
+    let err = TransactionBuilder::mint(
+        block_height,
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+    )
+    .finalize()
+    .check(block_height, &test_params())
+    .expect_err("Expected erroneous transaction");
 
-    assert_eq!(err, CheckError::TransactionMintOutputIsNotCoin);
+    assert_eq!(err, CheckError::TransactionMintIncorrectOutputIndex);
```

Comment:
nit: This has the (very small) chance of failing intermittently, since the `input_index` is random and could land on 0. I think it could be nice to set up the test in a way that shows why this is error is expected.